keras_cv/models/object_detection/retina_net/retina_net.py

```diff
@@ -222,34 +244,21 @@ def _forward(self, images, training=None):
 
         cls_pred = tf.concat(cls_pred, axis=1)
         box_pred = tf.concat(box_pred, axis=1)
-
-        return box_pred, cls_pred
-
-    def call(self, images, training=None):
-        if isinstance(images, tf.RaggedTensor):
-            raise ValueError(
-                "`RetinaNet()` does not yet support inputs of type `RaggedTensor` for input images. "
-                "To correctly resize your images for object detection tasks, we recommend resizing using "
-                "`keras_cv.layers.Resizing(pad_to_aspect_ratio=True, bounding_box_format=your_format)`"
-                "on your inputs."
-            )
-        box_pred, cls_pred = self._forward(images, training=training)
-        if not training:
-            # box_pred is on "center_yxhw" format, convert to target format.
-            anchors = self.anchor_generator(images[0])
-            anchors = tf.concat(tf.nest.flatten(anchors), axis=0)
-            box_pred = _decode_deltas_to_boxes(
-                anchors=anchors,
-                boxes_delta=box_pred,
-                anchor_format=self.anchor_generator.bounding_box_format,
-                box_format=self.bounding_box_format,
-                variance=BOX_VARIANCE,
-            )
         return box_pred, cls_pred
 
     def decode_predictions(self, predictions, images):
         # no-op if default decoder is used.
         box_pred, cls_pred = predictions
+        # box_pred is on "center_yxhw" format, convert to target format.
+        anchors = self.anchor_generator(images[0])
```

Comment:
Is it safe to assume images aren't ragged wrt. size?

Reply:
Good catch - I used to have a check for this in call and accidentally removed it here.  I've added it back to the call method to make sure that we catch this as soon as possible (the moment train step or eval step starts)